crates/view/src/storage.rs

```diff
@@ -1493,4 +1500,54 @@ impl Storage {
         })
         .await?
     }
+
+    pub async fn amount_by_sender(
+        &self,
+        start_height: Option<u64>,
+        end_height: Option<u64>,
+    ) -> anyhow::Result<BTreeMap<(Address, String), u128>> {
```

Comment:
Why do we want this to do the summarization internally? Wouldn't it be more flexible to have something like
```
notes_by_sender(&self, return_address: &Address) -> Result<Vec<SpendableNoteRecord>>
```
and then let the caller figure out how to interpret the results?

Reply:
Well, we want the signature to be such that the output contains all the addresses, rather than us having to figure out what addresses we need to pass into the function. This would be a return type of like `Iterator<(Address, SpendableNoteRecord)>` with the address appearing multiple times.

While this might be useful in more situations, right now the single consumer of this API just wants the total amount received. I think we can just punt the issue of providing a more generally useful version of the API once there's more consumers of that API.

Reply:
Okay actually I think implementing Henry's suggestion makes sense now :p

Reply:
The key point is that for the coordinator we're making this API for, we only ever want to query for single, specific addresses, and we don't want to have to scan through a potentially unbounded amount of data to run the query.